src/routes/organization/join/[organizationId]/+page.server.ts

```diff
@@ -0,0 +1,42 @@
+import { checkSessionAuth, getUserOrganizations, readPath, weaklyAuthenticate, writePath } from '$lib/server/firebaseUtils';
+import { error, redirect } from '@sveltejs/kit';
+import type { Actions, PageServerLoad } from './$types';
+
+export const load: PageServerLoad = async ({ params: { organizationId }, cookies, url }) => {
+  const { uid } = await checkSessionAuth(cookies, { loginRedirect: `/organization/join/${organizationId}` });
+  const publicOrgDetails = await readPath<Database.Organization.Public>(`/organizations/${organizationId}/public`);
+  if (!publicOrgDetails) throw error(404);
+
+  // Before proceeding, check if the user is invited to the org, and if so redirect them to the invite page.
+  const existingInviteKeys = await readPath<string[]>(`/organizations/${organizationId}/private/invites/`, []);
+  if (existingInviteKeys) {
+    await Promise.all(
+      existingInviteKeys.map(async (key) => {
+        const invite = await readPath<Database.Invite.Organization>(`/invites/organization/${key}`);
+        if (invite?.inviteeUid === uid) throw redirect(302, `/organization/invite/${key}`);
+      }),
+    );
+  }
+
+  const existingRequestPromise = readPath<Database.Organization.InviteRequest>(`/organizations/${organizationId}/private/inviteRequests/${uid}`);
+  const existingMemberPromise = getUserOrganizations(uid).then((orgs) => orgs.includes(organizationId));
+  const [existingRequest, existingMember] = await Promise.all([existingRequestPromise, existingMemberPromise]);
+
+  const message = url.searchParams.get('message');
```

Comment:
Having a query param for message will open up some additional possibilities like links that pre-populate the message with their source